configuration/configmgr/configmgrlib/mod_templates/IPAddressRangeInput.cpp

```diff
@@ -0,0 +1,164 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "IPAddressRangeInput.hpp"
+#include "TemplateException.hpp"
+#include "Utils.hpp"
+
+void IPAddressRangeInput::setValue(const std::string &range)
+{
+    //
+    // Formats accepted:
+    //  iplist = ip[;ip]
+    //  ip = a.b.c.d | a.b.c.g-h | hostname
+
+    m_values.clear();
+    std::vector<std::string> ipDefs = splitString(range, ";");
+    for (auto &ipDef: ipDefs)
+    {
+        std::string def = trim(ipDef);
+        bool isValid = true;
+
+        //
+        // Figure out what this def entails. Look at the number of parts. If one part, just
+        // save it.
+        std::vector<std::string> addressParts = splitString(def, ".");
+        if (addressParts.size() < 4)
```

Comment:
This would treat partial IP addresses as a hostname. IE: 127.0.0 would give us value=127

Reply:
True. The intent here is to support an IP address or range. The hostname capability should be removed and relegated to a separate input type. Trying to do too much. I shall remove the hostname from this class and add a new class.